src/visitors/transpileCssProp.js

```diff
@@ -49,7 +49,7 @@ export default t => (path, state) => {
   let injector
 
   if (TAG_NAME_REGEXP.test(name)) {
```

Comment:
This check seems fishy to me. Wouldn't we want to identify from the AST whether the element name is a JSX identifier or element name? Or does the AST not offer any hints on checking that?

Reply:
There isn't a clear understanding of if it's a scope variable or not, for example:

```
 Node {
        type: 'JSXIdentifier',
        start: 147,
        end: 150,
        loc:
         SourceLocation {
           start: Position { line: 7, column: 3 },
           end: Position { line: 7, column: 6 } },
        name: 'div' }
```

```
Node {
        type: 'JSXIdentifier',
        start: 1518,
        end: 1524,
        loc:
         SourceLocation {
           start: Position { line: 104, column: 34 },
           end: Position { line: 104, column: 40 } },
        name: 'Thing3' }
```

If you go and try to infer from local scope bindings there are also false positives:

```js
 console.log(
    path.scope.hasBinding(getVariableName(elem.node.name, t)),
    getVariableName(elem.node.name, t)
  )
```

```
false 'Button'
    console.log src/visitors/transpileCssProp.js:61
      false 'Button.Ghost'
    console.log src/visitors/transpileCssProp.js:61
      false 'button'
    console.log src/visitors/transpileCssProp.js:61
      false 'button-ghost'
    console.log src/visitors/transpileCssProp.js:61
      true 'p'
```

There's some variable called "p" in scope which conflicts with the tag named "p" and would trigger `styled(p)` which is wrong in this case.